### PR TITLE
Knowledge Adjustment Implementation

### PR DESCRIPTION
**Pull request by NeelayS**
- #44 

-  Changed loss_fn in attention and teacher_free to take a torch module

-  Added a check to confirm that a directory is present to save models. If not, creates and then saves.

### CODE REVIEW

**Comment by Het-Shah** [comment]
Do you want to add more or should I merge?

**Comment by NeelayS** [comment]
Actually, the paper suggests one more method. But that method would take a really long time to train because it calculates the loss one sample at a time. Plus, it also seems to be really hyperparameter sensitive. 
Doesn't add too much in my opinion.

**Comment by NeelayS** [comment]
Other than that, this is good to merge.

**Comment by Het-Shah** [comment]
Cool merging then.
